backend/src/main/scala/com/softwaremill/adopttapir/starter/content/ContentMerger.scala

```diff
@@ -0,0 +1,45 @@
+package com.softwaremill.adopttapir.starter.content
+
+object ContentMerger {
+
+  def createTree(topDirName: String, pathNames: List[String], content: String): Directory = {
+    assert(pathNames.nonEmpty, "`pathNames` list cannot be empty!")
+
+    def innerCreateTree(pathNames: List[String], content: String): Node = {
+      pathNames match {
+        case Nil          => throw new IllegalStateException("pathNames list cannot be empty!")
+        case last :: Nil  => File(last, content)
+        case head :: tail => Directory(head, List(innerCreateTree(tail, content)))
+      }
+    }
+
+    val pathsWithRootDir = List(topDirName) ++ pathNames
+    innerCreateTree(pathsWithRootDir, content).asInstanceOf[Directory]
```

Comment:
Instead of `.asInstanceOf`, maybe perform a `match` throwing an meaningful exception if a top-level file is found?

Reply:
This is a very nice suggestion. Thanks! 👍